### DISCUSSION TITLE
[提问] 如何实现会话超时提醒？

### DISCUSSION THREAD

**Discussion by Nattsu39** [nonebot/discussions#18] [问答] [answered]
如果用户没有在[session-expire-timeout](https://v2.nonebot.dev/api/config.html#session-expire-timeout)设定的时间内进一步操作则运行函数并提醒。
翻了下文档没找到相关内容..

**Comment by mnixry** [comment, marked as answer]
我能想到的是用事件处理函数中执行 [`loop.call_later`](https://docs.python.org/zh-cn/3/library/asyncio-eventloop.html#asyncio.loop.call_later) 来添加一个在到达超时事件后执行的回调

但是不知道怎么样在正常响应的情况下取消回调(

**Reply by stdrc** [reply]
call_later 跑的 callback 先检查一下某个状态或 flag（
